google-home-community.groovy

```diff
@@ -1774,10 +1798,27 @@ private executeCommand_volumeRelative(deviceInfo, command) {
     def volumeChange = command.params.relativeSteps
     def device = deviceInfo.device
 
-    def currentVolume = device.currentValue(volumeTrait.volumeAttribute)
-    // volumeChange will be negative when decreasing volume
-    def newVolume = currentVolume + volumeChange
-    device."${volumeTrait.setVolumeCommand}"(newVolume)
+    def newVolume
+
+    if (volumeTrait.canSetVolume) {
+        def currentVolume = device.currentValue(volumeTrait.volumeAttribute)
+        // volumeChange will be negative when decreasing volume
+        newVolume = currentVolume + volumeChange
+        device."${volumeTrait.setVolumeCommand}"(newVolume)
+    } else {
+        def volumeChangeCommand = volumeTrait.volumeUpCommand
+        if (volumeChange < 0) {
+            volumeChangeCommand = volumeTrait.volumeDownCommand
+        }
+
+        device."${volumeChangeCommand}"()
+        for (int i = 1; i < volumeTrait.volumeStep; i++) {
+            pauseExecution(100)
+            device."${volumeChangeCommand}"()
+        }
```

Comment:
Should this all be in a loop to handle the case when `volumeChange` has an absolute value greater than one?

Reply:
The reason for looping is to add the delay.
I didn't want the first call to be in the loop so that if it's just 1, we don't use any delay, and the delay kicks in for before consequent calls only.

Does that answer the question?

If you see any better way, I'd love to hear your suggestions 🙏

Reply:
The loop that's here looks fine, but I think you need another outer loop.  Someone can say something like "Hey Google, turn up the TV a lot" and `relativeSteps` field will be greater than one, so I think this whole loop should be inside another loop that runs multiple times based on that field.

Reply:
Great catch!
I didn't figure out that you could ask google to "increase volume by X".
Handling that did not require another loop since `relativeSteps` already takes in account the step.
I update the behavior here to match the previous implementation, buy using this value to decide on the amount of steps.

Tested thoroughly (how about an infinite loop increasing the TV volume at some point?) and now all cases are accurately handled. (both "Inc/dec volume" and "Inc/dec by X")